README.md

```diff
@@ -103,7 +103,54 @@ OpenShift LightSpeed (OLS) is an AI powered assistant that runs on OpenShift and
       ```
       **Note:** using static token will require you to set the `k8s_cluster_api` mentioned in section 6.4, as this will disable the loading of OCP config from in-cluster/kubeconfig.
 
-7. (Optional) Configure the document store
+7. Configure OLS TLS communication
+
+   This section provides instructions on configuring TLS (Transport Layer Security) for the OLS Application, enabling secure connections via HTTPS. TLS is enabled by default; however, if necessary, it can be disabled through the `dev_config` settings.
+
+
+   1. Enabling and Disabling TLS
+   
+      By default, TLS is enabled in OLS. To disable TLS, adjust the `dev_config` in your configuration file as shown below:
+
+      ```yaml
+         dev_config:
+            disable_tls: false
+      ```
+
+   2. Configuring TLS in local Environments:
+
+      1. Generate Self-Signed Certificates: To generate self-signed certificates, run the following command from the project's root directory:
+         ```bash
+            ./scripts/generate-certs.sh
+         ``` 
+      2. Update OLS Configuration: Modify your config.yaml to include paths to your certificate and its private key:
+         ```yaml
+            ols_config:
+               tls_config:
+                  ssl_certfile: /full/path/to/certs/cert.pem
+                  ssl_keyfile: /full/path/to/certs/key.pem
+         ```
+      3. Launch OLS with HTTPS: After applying the above configurations, OLS will run over HTTPS.
+   
+   3. Configuring OLS in OpenShift:
+
+      For deploying in OpenShift, Service-Served Certificates can be utilized. Update your ols-config.yaml as shown below, based on the example provided in the examples directory:
+
+      ```yaml
+         ols_config:
+            tls_config:
+               ssl_certfile: /app-root/certs/cert.pem
+               ssl_keyfile: /app-root/certs/key.pem
```

Comment:
yes, but this is the term being used by uvicorn
we can replace it with tls, but this will require us to change the usage of kwargs (i used same name of the parameters names being used by uvicorn)


            uvicorn.run(
                "ols.app.main:app",
                host=host,
                port=8443,
                workers=1,
                **config.ols_config.tls_config.__dict__,
            )

Reply:
uvicorn is an implementation detail, so our config should not be tied to how uvicorn expects config arguments.

Reply:
Resolved